src/Tools/ExternalAccess/VisualDiagnostics/Internal/VisualDiagnosticsServiceFactory.cs

```diff
@@ -0,0 +1,89 @@
+
+/// <summary>
+/// LSP Service responsible for loading IVisualDiagnosticsLanguageService workspace service and delegate the broker service to the workspace service,
+/// and handling MAUI XAML/C#/CSS/Razor Hot Reload support
+/// </summary>
+[Export(typeof(IOnServiceBrokerInitialized))]
+[ExportCSharpVisualBasicLspServiceFactory(typeof(OnInitializedService)), Shared]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+[method: ImportingConstructor]
+internal sealed class VisualDiagnosticsServiceFactory(
+    LspWorkspaceRegistrationService lspWorkspaceRegistrationService) : ILspServiceFactory, IOnServiceBrokerInitialized
+{
+    private readonly LspWorkspaceRegistrationService _lspWorkspaceRegistrationService = lspWorkspaceRegistrationService;
+    private readonly Lazy<OnInitializedService> _OnInitializedService = new Lazy<OnInitializedService>(() => new OnInitializedService(lspWorkspaceRegistrationService));
+
+    public ILspService CreateILspService(LspServices lspServices, WellKnownLspServerKinds serverKind)
+    {
+        return _OnInitializedService.Value;
+    }
+
+    public void OnServiceBrokerInitialized(IServiceBroker serviceBroker)
+    {
+        _OnInitializedService.Value.OnServiceBrokerInitialized(serviceBroker);
+    }
+
+    private class OnInitializedService : ILspService, IOnInitialized, IOnServiceBrokerInitialized, IDisposable
+    {
+        private readonly LspWorkspaceRegistrationService _lspWorkspaceRegistrationService;
+        private IVisualDiagnosticsLanguageService? _visualDiagnosticsLanguageService;
+        private CancellationToken _cancellationToken;
+        private static readonly TaskCompletionSource<bool> _taskCompletionSource = new TaskCompletionSource<bool>();
+
+        public OnInitializedService(LspWorkspaceRegistrationService lspWorkspaceRegistrationService)
+        {
+            _lspWorkspaceRegistrationService = lspWorkspaceRegistrationService;
+        }
+
+        public void Dispose()
+        {
+            (_visualDiagnosticsLanguageService as IDisposable)?.Dispose();
+        }
+
+        public Task OnInitializedAsync(ClientCapabilities clientCapabilities, RequestContext context, CancellationToken cancellationToken)
+        {
+            _cancellationToken = cancellationToken;
+            _taskCompletionSource.SetResult(true);
+            return Task.CompletedTask;
+        }
+
+        public void OnServiceBrokerInitialized(IServiceBroker serviceBroker)
+        {
+            _taskCompletionSource.Task.ContinueWith((initialized) => OnInitializeVisualDiagnosticsLanguageServiceAsync(serviceBroker), TaskScheduler.Default);
+        }
+
+        private async Task OnInitializeVisualDiagnosticsLanguageServiceAsync(IServiceBroker serviceBroker)
+        {
+            // initialize VisualDiagnosticsLanguageService
+            Workspace workspace = _lspWorkspaceRegistrationService.GetAllRegistrations().Where(w => w.Kind == WorkspaceKind.Host).FirstOrDefault();
```

Comment:
You don't need a separate `Where` clause.

```suggestion
            Workspace workspace = _lspWorkspaceRegistrationService.GetAllRegistrations().FirstOrDefault(w => w.Kind == WorkspaceKind.Host);
```

Reply:
Actually it should be first.  There's always going to be a host workspace